rsocket/transports/tcp/TcpDuplexConnection.cpp

```diff
@@ -192,10 +192,15 @@ class TcpOutputSubscriber
     tcpReaderWriter_->setOutputSubscription(nullptr);
   }
 
-  void onError(std::exception_ptr ex) noexcept override {
+  void onError(std::exception_ptr) noexcept override {
     CHECK(tcpReaderWriter_);
     auto tcpReaderWriter = std::move(tcpReaderWriter_);
-    tcpReaderWriter->closeErr(folly::exception_wrapper(std::move(ex)));
+    // FIXME: See https://github.com/rsocket/rsocket-cpp/issues/589
+    // The expensive alternative is to throw and the get the exception
+    // reference.  Since the message is internal only (not sent over wire),
+    // discarding the message (ex) as a short term fix to get things working in
+    // a debug build.  We should move to one type soon.
+    tcpReaderWriter->closeErr(folly::exception_wrapper());
```

Comment:
Commenting here again, I would prefer us to throw for now instead of changing behavior.